src/swap/SwapReviewScreen.tsx

```diff
@@ -0,0 +1,331 @@
+import networkConfig from 'src/web3/networkConfig'
+import { accountAddressSelector } from 'src/web3/selectors'
+
+type Props = StackScreenProps<StackParamList, Screens.SwapReviewScreen>
+
+const TAG = 'SWAP_REVIEW_SCREEN'
+
+export function SwapReviewScreen(props: Props) {
+  const { toToken, fromToken, swapAmount, updatedField } = props.route.params
+  const [swapInfo, setSwapInfo] = useState(null as any)
+  const [loading, setLoading] = useState(false)
+  const [fetchSwapQuoteError, setFetchSwapQuoteError] = useState(false)
+  const [estimatedModalVisible, setEstimatedDialogVisible] = useState(false)
+  const coreTokens = useSelector(coreTokensSelector)
+  const maxSlippagePercent = useSelector(maxSwapSlippagePercentageSelector)
+  const walletAddress = useSelector(accountAddressSelector)
+  const maxSlippageDecimal = `${maxSlippagePercent / 100}`
+
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+
+  // Token Symbols
+  const toTokenSymbol = coreTokens.find((token) => token.address === toToken)?.symbol
+  const fromTokenSymbol = coreTokens.find((token) => token.address === fromToken)?.symbol
+
+  useEffect(() => {
+    if (fetchSwapQuoteError) {
+      dispatch(showError(ErrorMessages.FETCH_SWAP_QUOTE_FAILED))
+      setTimeout(() => {
+        navigateBack()
+      }, 300)
+    }
+  }, [fetchSwapQuoteError])
+
+  useEffect(() => {
+    ValoraAnalytics.track(SwapEvents.swap_review_screen_open, {
+      toToken,
+      fromToken,
+      swapAmount,
+    })
+
+    void getSwapInfo().then((data) => {
+      setSwapInfo(data)
+    })
+  }, [])
+
+  const getSwapInfo = async () => {
+    try {
+      setLoading(true)
+      const swapAmountInWei = multiplyByWei(swapAmount[updatedField]!)
+      const swapAmountParam = updatedField === Field.FROM ? 'sellAmount' : 'buyAmount'
+      const params = {
+        buyToken: toToken,
+        sellToken: fromToken,
+        [swapAmountParam]: swapAmountInWei.toString().split('.')[0],
+        slippagePercentage: maxSlippageDecimal,
+        userAddress: walletAddress,
+      }
+      const requestUrl = `${networkConfig.approveSwapUrl}?${qs.stringify(params)}`
+      const response = await fetch(requestUrl)
+      if (!response.ok) {
+        Logger.error(TAG, `Failure response fetching token Swap: ${response}`)
+        throw new Error(
+          `Failure response fetching token swap quote. ${response.status}  ${response.statusText}`
+        )
+      }
+      return await response.json()
+    } catch (error) {
+      setFetchSwapQuoteError(true)
+      Logger.error(TAG, `Error fetching token swap quote: ${error}`)
+    } finally {
+      setTimeout(() => {
+        setLoading(false)
+      }, 300)
+    }
+  }
+
+  return (
+    <SafeAreaView style={styles.safeAreaContainer}>
+      <CustomHeader
+        title={t('swapReviewScreen.title')}
+        left={<BackButton />}
+        // handle large text size to avoid overlaps in header
+        right={
```

Comment:
maybe nice to check with Nitya one more time about whether we need this right button at all, as it doesn't exist in any of the other flows that are triggered from the drawer menu....

Reply:
Checked with Nitya and was removed in https://github.com/valora-inc/wallet/pull/2767/commits/70960e345f0debe2cbefb3fce5e70ef19e3fddca and https://github.com/valora-inc/wallet/pull/2767/commits/af577274eaabef8067b4a9f956e4242da13f1df2!